pkg/semtech-loramac/contrib/semtech_loramac_timer.c

```diff
@@ -60,7 +60,7 @@ void TimerSetValue(TimerEvent_t *obj, uint32_t value)
         xtimer_remove(&(obj->dev));
     }
 
-    obj->timeout = (value - 50) * 1000;
+    obj->timeout = (value - 20) * 1000;
```

Comment:
BTW what sorcery is this number?

Reply:
This comes from the specs: the RX windows start RX_DELAY seconds with +/-20ms error, so anticipating the RX window with 20ms should ensure that a message from the gateway is not missed.
The initial port was not very accurate : the RX windows were starting is little too late, hence missing the gateway message preamble, that's why I anticipated them a little more (hence the arbitrary 50ms value), probably because of thread switching overheads. But with this PR, it behaves as expected, and it works with 20ms.

Reply:
Could you add a little comment above this line to explain it please ? I don't think it is a good idea to have hardcoded number in the wild without any tips.

Reply:
> Could you add a little comment above this line to explain it please ?

that's a good idea (and practice), I'll change that.